apps/dav/lib/CalDAV/Calendar.php

```diff
@@ -63,6 +65,11 @@ class Calendar extends \Sabre\CalDAV\Calendar implements IRestorable, IShareable
 	 * @param IConfig $config
 	 */
 	public function __construct(BackendInterface $caldavBackend, $calendarInfo, IL10N $l10n, IConfig $config) {
+		// Convert deletion date to ISO8601 string
+		if (isset($calendarInfo[\OCA\DAV\CalDAV\Trashbin\Plugin::PROPERTY_DELETED_AT])) {
+			$calendarInfo[\OCA\DAV\CalDAV\Trashbin\Plugin::PROPERTY_DELETED_AT] = (new DateTimeImmutable('@' . $calendarInfo[\OCA\DAV\CalDAV\Trashbin\Plugin::PROPERTY_DELETED_AT]))->format(DateTimeInterface::ATOM);
```

Comment:
```suggestion
			$calendarInfo[\OCA\DAV\CalDAV\Trashbin\Plugin::PROPERTY_DELETED_AT] = (new DateTimeImmutable())->setTimestamp($calendarInfo[\OCA\DAV\CalDAV\Trashbin\Plugin::PROPERTY_DELETED_AT]))->format(DateTimeInterface::ATOM);
```
Maybe cleaner. No big deal.